### PR TITLE
Fix #109661: Regression: Page settings does not work for parts

### PR DESCRIPTION
**Pull request by neGjodsbol**
Cause of problem was achange to call of setScore within musescore/showPagesettings.
The call was changed from setScore(cs) to  setScore(cs->masterScore) caused by cs no longer being of type MasterScore.
This resulted in all changes in PageSettings to be applied to master even if called from a part.

Type of parameter in setScore changed to Score* and call changed to  setScore(cs)

Cast to MasterScore added within pagesettings/setScore before cloning.

### CODE REVIEW

**Comment by anatoly-os** [comment]
@neGjodsbol could you please remove all these "merge" commits and just use rebasing?
Let me know if you need help with it.

File: mscore/pagesettings.cpp [hunk 1]
```diff
@@ -97,10 +97,10 @@ void PageSettings::hideEvent(QHideEvent* ev)
 //   setScore
 //---------------------------------------------------------
 
-void PageSettings::setScore(MasterScore* s)
+void PageSettings::setScore(Score* s)
       {
-      cs  = s;
-      MasterScore* sl = s->clone();
+	  cs = static_cast<MasterScore*>(s);
```
Comment (Jojo-Schmitz) [inline mscore/pagesettings.cpp]: whitespace damage, we indent by 6 spaces not tabs

Comment (neGjodsbol) [reply mscore/pagesettings.cpp]: Noted. First time for me.

Will get to work in these two

**Comment by neGjodsbol** [comment]
Ran into trouble with squash.
Created new PR #3948, with only one commit.

**Comment by anatoly-os** [comment]
@neGjodsbol this solution is the best solution, but it requires more changes. I will prepare the patch with proposed changes and give it to you, so you can check all scenarios and fix the code if necessary. Give me a day.

**Pull request closed without merging**